Lib/_pydecimal.py

```diff
@@ -500,7 +500,13 @@ def sin(x):
     >>> print(getcontext().prec)
     28
     """
-    if ctx is None: ctx = getcontext()
+    if ctx is None:
+        ctx = getcontext()
+    for key, value in kwargs.items():
+        try:
+            setattr(ctx, key, value)
+        except:
```

Comment:
Don't use `except:` since it will catch even things like KeyboardInterrupt. We should catch only the specific exception that `setattr()` can throw, which in this case is `AttributeError`. We could also just hardcode the list of allowed attributes and check it directly.